src/plugins/cesium/ngv-plugin-cesium-offline.ts

```diff
@@ -184,18 +188,24 @@ export class NgvPluginCesiumOffline extends LitElement {
         const tilesetName = splitted[1];
         const catalog = catalogs.find((c) => c.id === id);
         if (catalog?.layers[tilesetName]?.type === '3dtiles') {
+          let accessToken: string | undefined;
           try {
+            let url = catalog.layers[tilesetName].url;
+            if (typeof url === 'number') {
+              const id = url;
+              url = `${ION_ASSETS_URL}${id}/`;
+              accessToken = await getIonAssetToken(id);
+            }
             await downloadAndPersistTileset({
               appName: this.info.appName,
               subdir: this.info.tiles3dSubdir,
               concurrency: 3,
-              tilesetBasePath: (<string>(
-                catalog.layers[tilesetName].url
-              )).replace('tileset.json', ''),
+              tilesetBasePath: url.replace('tileset.json', ''),
```

Comment:
Indeed, we need to rework the whole URL resolution mechanism.
We need to rely on `resources` and not URLs like I did. It will probably simplify our code.

Reply:
It looks like the derived token is valid for 1h.
So your solution is OK, no need to refactor everything to use resources.